### DISCUSSION TITLE
Does BlenderBIM support IFC 4x3 RC?

### DISCUSSION THREAD

**Discussion by cm-wsp** [IfcOpenShell/IfcOpenShell#2131] [Q&A]
Hi,
for the last couple of months I've been trying to get a grip on Blender the BlenderBIM Add On and the OS Workflows. I work mainly on Roads and Rail infrastructure and was wondering if Blender BIM supports the IFC 4x3 RC schema. Because IFCOpenshell, in my understanding, does.

I know RC means release candidate, and I'm slowly gaining confidence with BlenderBIM, but when I try to assign properties and classes to an element I only see the ones available for IFC 4, I even didn't find the Alignment properties or elements of IFC 4.1. Is this correct or am I missing something?

Best regards,
CarlosM.

**Comment by aothms** [comment]
There's three layers of support for schema definitions:

- compiled (early bound) support for the schema: 2x3 4 4x1 4x2 4x3_rc1 4x3_rc2 4x3_rc3 4x3_rc4 https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/cmake/CMakeLists.txt#L582
- support for the geometry definitions in the schema: there is a (probably no longer up to date list of items implemented here https://github.com/IfcOpenShell/IfcOpenShell/issues/59) support for the entities from 4x1 upwards is mostly entirely lacking. Note that 4x1 and 4x2 are withdrawn as bSI standards.
- blender bim schema handling which appears to be mostly hardcoded to IFC4 https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/blenderbim/blenderbim/bim/schema.py

So there is support in terms of parsing, but no support for interpreting the geometry to visible elements. Some exploratory implementation has started in Python https://github.com/IfcOpenShell/IfcOpenShell/pull/1470 https://github.com/IfcOpenShell/IfcOpenShell/pull/1480 

Contributions are very welcome :) Let me know if you want to know more.
